src/components/Header/RightNav.jsx

```diff
@@ -25,19 +27,46 @@ const Ul = styled.ul`
 
     li {
       color: #fff;
+      cursor: pointer;
     }
   }
 `;
 
 const RightNav = ({ open }) => {
+  // created only for being able to impliment the if
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
   return (
-    <Ul open={open}>
-        <li>Home</li>
-        <li>About Us</li>
-        <li>Contact Us</li>
-        <li>Sign In</li>
-        <li>Sign Up</li>
-    </Ul>
+    <>
+      
+      {/* map through the array with routes and route names and returning the correspondig page*/}
+      {/* map through the array with an authenticated and unauthenticated user and returning the correspondig page*/}
+
+      {isLoggedIn ? (
+        forLoggedInUser.map(function(user) {
```

Comment:
You're super close! Remember that the argument of the map callback function will represent whatever is in the array you are iterating over. So, since forLoggedInUser contains an array of route objects, mapping over it would look more like this:

forLoggedInUser.map(function(routeObj) {
}

then, you can access the route names contained in those routeObjs for your <li> elements instead of hard coding them. Like this:
...map function.... {
      <li key={routeObj.name}>{routeObj.name}<li>
}

this will create a <li> element for each route object in the array you're mapping over. 

the key attribute on the <li> element is required by react anytime you render the results of a map function. It has to be a unique value, so the name of the route works just fine here. 

Once pages are set up for the different menu routes we will need to come back to this code and wrap the route names in a <Link> or <a> tag so that users can click them to navigate to the different pages of the app. 


forSignedOutUser works exactly the same way. 

I hope that makes sense! If not, we can pair on it tomorrow- just ping me in the channel!